distribution.md

```diff
@@ -0,0 +1,180 @@
+# Distribution Specification
+
+This document specifies the artifact format, delivery mechanism, and order resolution process for buildpacks.
+
+
+## Table of Contents
+
+1. [Order Resolution](#order-resolution)
+2. [Artifact Format](#artifact-format)
+   1. [Buildpack Blob](#buildpack-blob)
+   2. [Buildpackage](#buildpackage)
+3. [Data Format](#data-format)
+   1. [buildpack.toml (TOML)](#buildpack.toml-toml)
+
+## Order Resolution
+
+During detection, a buildpack ID or buildpack order definition MUST be resolved into individual buildpack implementations.
+
+The resolution process MUST follow this pattern:
+
+Where:
+- O and P are buildpack orders.
+- A through H are buildpack implementations.
+
+Given:
+
+<img src="http://tex.s2cms.ru/svg/%0AO%20%3D%0A%5Cbegin%7Bbmatrix%7D%0AA%2C%20%26%20B%20%5C%5C%0AC%2C%20%26%20D%0A%5Cend%7Bbmatrix%7D%0A" alt="
+O =
+\begin{bmatrix}
+A, &amp; B \\
+C, &amp; D
+\end{bmatrix}
+" />
+
+<img src="http://tex.s2cms.ru/svg/%0AP%20%3D%0A%5Cbegin%7Bbmatrix%7D%0AE%2C%20%26%20F%20%5C%5C%0AG%2C%20%26%20H%0A%5Cend%7Bbmatrix%7D%0A" alt="
+P =
+\begin{bmatrix}
+E, &amp; F \\
+G, &amp; H
+\end{bmatrix}
+" />
+
+We propose:
+
+<img src="http://tex.s2cms.ru/svg/%0A%5Cbegin%7Bbmatrix%7D%0AE%2C%20%26%20O%2C%20%26%20F%0A%5Cend%7Bbmatrix%7D%20%3D%20%0A%5Cbegin%7Bbmatrix%7D%0AE%2C%20%26%20A%2C%20%26%20B%2C%20%26%20F%20%5C%5C%0AE%2C%20%26%20C%2C%20%26%20D%2C%20%26%20F%20%5C%5C%0A%5Cend%7Bbmatrix%7D%0A" alt="
+\begin{bmatrix}
+E, &amp; O, &amp; F
+\end{bmatrix} = 
+\begin{bmatrix}
+E, &amp; A, &amp; B, &amp; F \\
+E, &amp; C, &amp; D, &amp; F \\
+\end{bmatrix}
+" />
+
+<img src="http://tex.s2cms.ru/svg/%0A%5Cbegin%7Bbmatrix%7D%0AO%2C%20%26%20P%0A%5Cend%7Bbmatrix%7D%20%3D%20%0A%5Cbegin%7Bbmatrix%7D%0AA%2C%20%26%20B%2C%20%26%20E%2C%20%26%20F%20%5C%5C%0AA%2C%20%26%20B%2C%20%26%20G%2C%20%26%20H%20%5C%5C%0AC%2C%20%26%20D%2C%20%26%20E%2C%20%26%20F%20%5C%5C%0AC%2C%20%26%20D%2C%20%26%20G%2C%20%26%20H%20%5C%5C%0A%5Cend%7Bbmatrix%7D%0A" alt="
+\begin{bmatrix}
+O, &amp; P
+\end{bmatrix} = 
+\begin{bmatrix}
+A, &amp; B, &amp; E, &amp; F \\
+A, &amp; B, &amp; G, &amp; H \\
+C, &amp; D, &amp; E, &amp; F \\
+C, &amp; D, &amp; G, &amp; H \\
+\end{bmatrix}
+" />
+
+Note that buildpack IDs are expanded depth-first in left-to-right order.
+
+If a buildpack order entry within a group has the parameter `optional = true`, then a copy of the group without the entry MUST be repeated after the original group.
+
+## Artifact Format
+
+### Buildpack Blob
+
+A buildpack blob MUST contain one or more buildpacks.
+
+A buildpack blob MUST be packaged as gzip-compressed tarball.
+Its filename should end in `.tgz`.
+
+A buildpack blob MUST contain a `buildpack.toml` file at its root directory.
+
+A buildpack defined within `buildpack.toml` MUST either be:
+
+1. A buildpack implementation specified by a `stacks` field or
```

Comment:
"specified by" almost makes it sound like the stack value will somehow point to the buildpack implementation. Wonder if there's a better wording here?

Reply:
Maybe: "denoted by the presence of a `stacks` field" or something like that?

Reply:
👍